# EDA with Python and Pandas using Forbes 2022 Dataset
![](./imgs/happy-rich-banker-celebrating-income-growth_74855-5867.jpg)

Olá! Seja bem-vindo(a) à nossa avaliação. 😀

Este é o momento de mostrar o que você aprendeu. Leia cada questão com calma, respire fundo e confie no seu conhecimento.

Estamos na torcida! Boa prova!

<b>Conteudo:</b>
<ul>
<li><a href="#load">Loading the dataset</a></li>  
<li><a href="#preprocessing">Data preprocessing</a></li>
<li><a href="#gender">Gender analysis</a></li>
<li><a href="#richest">Top 10 richest in the world</a></li>
<li><a href="#countries">Top 10 countries</a></li>
<li><a href="#categories">Top 10 categories</a></li>   
<li><a href="#age">The distribution of age</a></li>        
</ul>

### Primeiros passos 
Abra um + code e instale as libs abaixo:

pip install pandas <br/>
pip install numpy <br/>
pip install matplotlib <br/>
pip intall seaborn <br/>

**Observação** um code para cada instalação acima

## Caso o seu python não funcione e necessite criar um ambiente Virtual(Python):

#### Abra o powershel e digite 
py --version

- Python 3.14.0 

#### Se a resposta for um valor parecido tipo (Python 3.13.2) : digite o comando abaixo:

py -m venv myvenv

#### digite o comando abaixo para ativar o mesmo

myvenv\Scripts\activate

- (myvenv) C:\Users\seuUser

#### vai ao Vscode e busque na lateral direita e clique na versão so Python e vai mostrar na barra acima selecione outro Kernel, navegue até a pasta myvenv C:\Users\seuUser\myvenv\Script e selecione o python.exe la dentro, restart seu ipynb

![myvenv](./imgs/myEnv.png)

# Loading the dataset 
[the Forbes world's billionaires list 2022 dataset](https://www.forbes.com/billionaires/)

Vamos carregar o conjunto de dados com o método `read_csv` do Pandas.

In [2]:
%pip install pandas
%pip install numpy

Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.


In [5]:
import pandas as pd
import numpy as np

In [8]:
#Olá! vamos ao  primeiro passo da nossa Análise Exploratória de Dados (EDA): 
# vamos carregar o arquivo usando o  read_csv do pandas,o arquivo "forbes_2022_billionaires.csv" no DataFrame df 
# e usamos df.head() para espiar as primeiras linhas.
# lembre-se e importante importar o pandas
# O arquivo está dentro da pasta datasets
df = pd.read_csv("./datasets/forbes_2022_billionaires.csv")

<a id="preprocessing"></a> 
# Data Preprocessing

A fase de compreensão é onde tentamos explorar os dados. Vamos dar uma olhada no formato do conjunto de dados com o atributo `shape`.

In [10]:
# mostre quantas linhas e colunas possuem o dataframe df criado na importação
df.shape

(2668, 22)

Como você pode ver, o conjunto de dados possui 2668 linhas e 22 colunas.

03. Deixe-me mostrar o tipo de colunas com o atributo `dtypes`.

In [11]:
# use o atributo dtypes do df principal
df.dtypes

rank                      int64
personName                  str
age                     float64
finalWorth              float64
year                      int64
month                     int64
category                    str
source                      str
country                     str
state                       str
city                        str
countryOfCitizenship        str
organization                str
selfMade                   bool
gender                      str
birthDate                   str
title                       str
philanthropyScore       float64
residenceMsa                str
numberOfSiblings        float64
bio                         str
about                       str
dtype: object

Observe que a coluna de classificação consiste em números. <br/>
Vamos converter essa coluna em uma coluna de índice com o método `set_index`.

In [ ]:
#Reset o index do dataframe df 
df.reset_index

#set o index do dataframe games usando 'rank' usando inplace=True 
df.set_index("rank", inplace=True)

#mostre os 5 primeiros registros
df.head(n=5)


,personName,age,finalWorth,year,month,category,source,country,state,city,...,organization,selfMade,gender,birthDate,title,philanthropyScore,residenceMsa,numberOfSiblings,bio,about
rank,,,,,,,,,,,,,,,,,,,,,
1,Elon Musk,50.0,219000.0,2022,4,Automotive,"Tesla, SpaceX",United States,Texas,Austin,...,Tesla,True,M,1971-06-28,CEO,1.0,NaN,NaN,Elon Musk is working to revolutionize transpor...,Musk was accepted to a graduate program at Sta...
2,Jeff Bezos,58.0,171000.0,2022,4,Technology,Amazon,United States,Washington,Seattle,...,Amazon,True,M,1964-01-12,Entrepreneur,1.0,"Seattle-Tacoma-Bellevue, WA",NaN,Jeff Bezos founded e-commerce giant Amazon in ...,"Growing up, Jeff Bezos worked summers on his g..."
3,Bernard Arnault & family,73.0,158000.0,2022,4,Fashion & Retail,LVMH,France,NaN,Paris,...,LVMH Moët Hennessy Louis Vuitton,False,M,1949-03-05,Chairman and CEO,NaN,NaN,NaN,Bernard Arnault oversees the LVMH empire of so...,"Arnault apparently wooed his wife, Helene Merc..."
4,Bill Gates,66.0,129000.0,2022,4,Technology,Microsoft,United States,Washington,Medina,...,Bill & Melinda Gates Foundation,True,M,1955-10-28,Cofounder,4.0,"Seattle-Tacoma-Bellevue, WA",NaN,Bill Gates turned his fortune from software fi...,"When Gates was a kid, he spent so much time re..."
5,Warren Buffett,91.0,118000.0,2022,4,Finance & Investments,Berkshire Hathaway,United States,Nebraska,Omaha,...,Berkshire Hathaway,True,M,1930-08-30,CEO,5.0,"Omaha, NE",NaN,"Known as the ""Oracle of Omaha,"" Warren Buffett...","Buffett still lives in the same Omaha, Nebrask..."


Analise os dados ausentes nas colunas com os métodos `isnull` e `sum`.

In [17]:
#responda aqui  Analise os dados ausentes nas colunas com os métodos `isnull` e `sum`. df principal
df.isnull().sum()

personName                 0
age                       86
finalWorth                 0
year                       0
month                      0
category                   0
source                     0
country                   13
state                   1920
city                      44
countryOfCitizenship       0
organization            2316
selfMade                   0
gender                    16
birthDate                 99
title                   2267
philanthropyScore       2272
residenceMsa            2029
numberOfSiblings        2541
bio                        0
about                   1106
dtype: int64

Como você pode ver, há dados faltantes nas colunas de idade, país e gênero. <br/> 
Como não há muitos dados faltantes nessas colunas, vamos removê-los com o método `dropna`.

In [18]:
# sua missão e remover os dadas faltantes em 'age'
# a função dropna possui um paramentro subset use 
# Use o 'inplace=True' outro paramentro
# O 'inplace=True' aplica a mudança diretamente no DataFrame 'df'.
df.dropna(subset=['age'], inplace=True)


Vamos analisar as informações gerais do conjunto de dados usando o método `info`.

In [19]:
# sua missão mostre as informações do dataframe principal
df.info

<bound method DataFrame.info of                     personName   age  finalWorth  year  month  \
rank                                                            
1                    Elon Musk  50.0    219000.0  2022      4   
2                   Jeff Bezos  58.0    171000.0  2022      4   
3     Bernard Arnault & family  73.0    158000.0  2022      4   
4                   Bill Gates  66.0    129000.0  2022      4   
5               Warren Buffett  91.0    118000.0  2022      4   
...                        ...   ...         ...   ...    ...   
2578             Zhang Yuqiang  66.0      1000.0  2022      4   
2578                Zhou Ruxin  59.0      1000.0  2022      4   
2578         Wen Zhou & family  57.0      1000.0  2022      4   
2578      Zhou Yifeng & family  43.0      1000.0  2022      4   
2578   Zhuang Kuilong & family  59.0      1000.0  2022      4   

                   category                   source        country  \
rank                                               

Como você pode ver, o conjunto de dados consiste em 2582 linhas e 6 colunas, e há dados faltantes sendo a maior quantidade em (numberOfSiblings) no conjunto de dados.

<a id="gender"></a>
# Gender analysis

Vamos analisar o gênero dos bilionários com o método `value_counts`.

In [20]:
# Use o campo 'gender' e verifique o metodo value_counts
df['gender'].value_counts

<bound method IndexOpsMixin.value_counts of rank
1       M
2       M
3       M
4       M
5       M
       ..
2578    M
2578    M
2578    M
2578    F
2578    M
Name: gender, Length: 2582, dtype: str>

A maioria dos bilionários são homens. Vamos analisar a porcentagem de homens e mulheres com o parâmetro `normalize=True` ativado.

In [21]:
# o metodo value_counts possui  um parametro normalize=True
# mostre a porcentagem do generos e não quantidade
df['gender'].value_counts(normalize=True)

gender
M    0.888199
F    0.111801
Name: proportion, dtype: float64

Como você pode ver, 89% dos bilionários são homens e 11% são mulheres. <br/> 
Vamos analisar as médias de idade por gênero. Para isso, vou usar o método `groupby`.

In [26]:
#Agrupe o mesmo e drie um novo daframe chamdado  df_gender
df_gender = df.groupby("gender")["age"]

Vamos calcular as idades médias com o método `mean`.

In [28]:
# use o metodo mean nas idade neste novo df (df_gender)
df_gender.head()
df_gender.mean()

gender
F    62.798611
M    64.392045
Name: age, dtype: float64

Como se pode observar, a idade média dos homens é de 64 anos e a das mulheres, de 63 anos. 
<br/>
Vamos criar um gráfico de barras para as médias masculina e feminina. Primeiro, deixe-me especificar um tema do Seaborn.

Vou usar o pacote `warnings` para evitar ver as mensagens de aviso.

In [29]:
import warnings
warnings.filterwarnings("ignore")

Vamos desenhar o gráfico de barras com o método `plot`.

<a id="richest"></a>
# Os 10 mais ricos do Brasil

In [30]:
# crie um novo df a partir do principal df 
# use o filtro de comparação e selecione todas as linhas 
# cujo o campo country for igual a Brazil
# crie um novo dataframe chamado df_country_brazil
# e mostre as 5 primeiras linhas 
df_country_brazil = df[df["country"] == "Brazil"]

Para ver a lista dos 10 mais ricos do mundo, vamos usar o método `barplot` do seaborn.

### 05 -  Questão Responda
### Descreva neste Markdown os dois mais ricos, visto pelo Dados gerado acima

<a id="countries"></a>
# Os 10 principais países

Vamos dar uma olhada nos países com o maior número de bilionários. <br/> 
Primeiro, mostre quantos países diferentes existem usando o método `len` .<br/>
Lembre-se de travar os paises com o metodo unique, use a celula abaixo

In [32]:
# # Use o df["country"] e use a função unique e conte quantos valores  tem usando a função global len(param)
df["country"].unique
len(df["country"].unique())

74

A lista contém 74 países. <br/>
Para criar um gráfico de barras com os dez países que possuem o maior número de bilionários.  <br/>
vamos agrupar os dados pela coluna "país" usando o método `groupby`.

In [34]:
#agrupe por country e crie um novo dataframe df_country
df_country = df.groupby("country").size()
df_country.head()

country
Algeria       1
Andorra       1
Argentina     4
Australia    40
Austria      12
dtype: int64

Vamos calcular o número de bilionários por país com o método `size`, e depois ordenar os países com o maior número de bilionários com `sort_values` e transformar esses dados em um dataframe com o método `DataFrame`.

In [ ]:
# use o df_country em ao executar  o comando solicitado acima
# coloque a resposta em um novo dataframe  chamado df_country_count
# use sort_values(ascending=False)
# pd.DataFrame(data, columns=["Count"])
# mostre a 5 primeiras linhas de df_country_count





Agora vamos desenhar um gráfico de barras para os dez primeiros países. <br/> Como você pode ver, o primeiro país com o maior número de bilionários são os Estados Unidos, seguidos pela China, etc.

<a id="categories"></a>
# Top 10 categories


Analise as categorias únicas com o método `unique`.

In [69]:
# use df principal e o campo category


Vamos remover os espaços das colunas e substituir `&` por `_` usando o método `replace` do str.

In [70]:
# use str.replace(" ","") no campo category removendo os espaços
# use novamente removendo o &

Vamos analisar novamente as categorias únicas com o método `unique`.

In [71]:
#use novamente 
# # use df principal e o campo category
 


Vamos descobrir o número de categorias. Para isso, primeiro usaremos o método `groupby` e depois o método `size`.

In [72]:
#crie uma nova dataframe chamado df_category
# agrupe por ategory e use a função size() em sequencia
#mostre as 5 primeiras linhas

Vamos converter esses dados em um dataframe.

In [73]:
#descomentar as duas linhas abaixo

#df_category = df_category.to_frame()
#df_category.head()

<a id="age"></a>
# A distribuição etária(age)

Vamos analisar a distribuição da coluna de idade.

In [74]:
#  Cria o histograma o sns possui um tipo de grafico que e  o histplot e sendo uma função 
# vc passa o dataframe principal e o campo a ser gerado no nosso caso age
# plt usando o mesmo do matplotlib use o metodo title para colocar o titulo do grafico The distribution of age

A idade dos mais ricos situa-se, em sua maioria, entre 50 e 80 anos.

### Mandou bem! 🚀

Parabéns por concluir a prova. Sabemos o quanto você se dedicou para chegar até aqui.

Agora é só respirar fundo e aguardar o resultado. Ótimo trabalho! 👍